pkg/controller/che/tls-secrets.go

```diff
@@ -0,0 +1,287 @@
+//
+// Copyright (c) 2020 Red Hat, Inc.
+// This program and the accompanying materials are made
+// available under the terms of the Eclipse Public License 2.0
+// which is available at https://www.eclipse.org/legal/epl-2.0/
+//
+// SPDX-License-Identifier: EPL-2.0
+//
+// Contributors:
+//   Red Hat, Inc. - initial API and implementation
+//
+package che
+
+import (
+	"context"
+	"time"
+
+	orgv1 "github.com/eclipse/che-operator/pkg/apis/org/v1"
+	"github.com/eclipse/che-operator/pkg/deploy"
+	"github.com/sirupsen/logrus"
+	batchv1 "k8s.io/api/batch/v1"
+	corev1 "k8s.io/api/core/v1"
+	rbac "k8s.io/api/rbac/v1"
+	"k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/controller/controllerutil"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// shouldReturn specifies whether caller of HandleCheTLSSecrets should return from reconcile loop.
+var shouldReturn = true
+
+// HandleCheTLSSecrets handles TLS secrets required for Che deployment
+func HandleCheTLSSecrets(checluster *orgv1.CheCluster, r *ReconcileChe) (bool, reconcile.Result, error) {
+	shouldReturn = true
+	reconcileResult, err := processCheTLSSecrets(checluster, r)
+	return shouldReturn, reconcileResult, err
+}
+
+func processCheTLSSecrets(checluster *orgv1.CheCluster, r *ReconcileChe) (reconcile.Result, error) {
+	var cheTLSSecretName = checluster.Spec.K8s.TlsSecretName
+	const cheTLSSelfSignedCertificateSecretName string = "self-signed-certificate"
+
+	// ===== Check Che TLS certificate ===== //
+
+	cheTLSSecret := &corev1.Secret{}
+	err := r.client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSSecretName}, cheTLSSecret)
+	if err != nil {
+		if !errors.IsNotFound(err) {
+			// Error reading secret info
+			logrus.Errorf("Error getting Che secert \"%s\": %v", cheTLSSecretName, err)
+			return reconcile.Result{}, err
+		}
+
+		// Che TLS secret doesn't exist, generate a new one
+
+		// Remove Che CA certificate secret if any
+		cheCASelfSignedCertificateSecret := &corev1.Secret{}
+		err = r.client.Get(context.TODO(), types.NamespacedName{Namespace: checluster.Namespace, Name: cheTLSSelfSignedCertificateSecretName}, cheCASelfSignedCertificateSecret)
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				// Error reading secret info
+				logrus.Errorf("Error getting Che secert \"%s\": %v", cheTLSSecretName, err)
```

Comment:
secert sounds not clear for me. Maybe `Self-Signed CA Certificate secret` ?

Reply:
Ok, will change to `Che TLS secret`. Works for you?